### DISCUSSION TITLE
Finetuning on custom dataset

### DISCUSSION THREAD

**Discussion by MartinMayoIglesias** [WongKinYiu/YOLO#191] [General]
Hello, I noticed that the notebook containing the training example is empty. I would like to use this repository to fine-tune YOLOv9 on a custom dataset.  I have issues to run the training, there seems to be a bug for Macbook computers

Just to be more precise: I am using MacBook, and I am executing the following:

python yolo/lazy.py task=train task.data.batch_size=8 model=v9-c dataset=handwritten_signatures device=cpu

with a yaml config file as explained in the repo which looks like

path: /Users/myname/Desktop/testing/signature_dataset
train: train
validation: val
class_num: 1
names: ["signature"]

and the dataset signature_dataset has the structure:

📂 signature_dataset
├── 📂 images
│   ├── 📂 train
│   └── 📂 val
└── 📂 labels
    ├── 📂 train
    └── 📂 val
    
which is a dataset parsed of of this dataset tech4humans/signature-detection from hugging face. So images contain .jpg while labels contain .txt

And after trying to train with the above bash command I get this error

/AppleInternal/Library/BuildRoots/d9889869-120b-11ee-b796-7a03568b17ac/Library/Caches/com.apple.xbs/Sources/MetalPerformanceShaders/MPSNDArray/Kernels/MPSNDArrayGatherND.mm:234: failed assertion `Rank of updates array (1) must be greater than or equal to inner-most dimension of indices array (8400)'
